src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -344,9 +344,12 @@ public async Task<ResultWrapper<Keccak>> eth_sendRawTransaction(byte[] transacti
         {
             try
             {
-                Keccak txHash =
+                (Keccak txHash, AddTxResult? addTxResult) =
                     await _txSender.SendTransaction(tx, txHandlingOptions | TxHandlingOptions.PersistentBroadcast);
-                return ResultWrapper<Keccak>.Success(txHash);
+
+                return addTxResult == AddTxResult.Added
+                    ? ResultWrapper<Keccak>.Success(txHash)
+                    : ResultWrapper<Keccak>.Fail(string.Concat(txHash, ": Transaction not added - ", addTxResult.ToString()), ErrorCodes.TransactionRejected);
```

Comment:
Can we check what error messages Geth returns and be more standard? Not sure if "Transaction not added -" prefix is adding any value, but not sure.

Reply:
Their example returns:

{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"unknown account"}}
{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"insufficient funds for gas * price + value"}}
{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"intrinsic gas too low"}}
{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"already known"}}
{"jsonrpc":"2.0","id":1,"result":"0x7a5a94d5b5e3ce017ce2c2022f02ec5db10611c43695c3256861bdb19317ab0e"}

so they are returning hash only when tx added, otherwise only error message. I think we can return only AddTxResult here, I will remove hash and prefix